### ISSUE TITLE
about batch normalization

### ISSUE THREAD

**Issue by neycyanshi** [tinghuiz/SfMLearner#30]
hi, you said BN is used for all layers except for output layers in paper, but I cannot find BN in code. As you explained in **Notes**, remove BN helps training, could you explain it or give me some insights?

**Comment by ClementPinard** [comment]
BN is usually good when using a large batch. 4 is a farily small batch, and a batch of only 4 elements will have a high variance of its mean and std. running_mean and running_std will have a hard time converging. In addition, based on what are the other 3 elements of a batch, the first element will potentially have a totally different normalization.

**Comment by dragonkao730** [comment]
Sorry to intrude, Can I ask ...
1. Have you ever tried batch renormalization?
2. The performance with BN and without BN?
Thank you!

**Comment by tinghuiz** [comment]
@dragonkao730 
1. By 'batch renormalization' I assume you meant this paper (https://arxiv.org/abs/1702.03275)? No, I have never tried it, but it seems like a good idea.

2. I tried BN with batch size = 4, and didn't seem to help. As @ClementPinard mentioned, this is likely due to the small batch size.
